giraf/binaryGiraf/binReader.go

```diff
@@ -0,0 +1,269 @@
+package binaryGiraf
+
+import (
+	"bytes"
+	"encoding/binary"
+	"fmt"
+	"github.com/biogo/hts/bgzf"
+	"github.com/vertgenlab/gonomics/cigar"
+	"github.com/vertgenlab/gonomics/common"
+	"github.com/vertgenlab/gonomics/dna"
+	"github.com/vertgenlab/gonomics/dnaThreeBit"
+	"github.com/vertgenlab/gonomics/fileio"
+	"github.com/vertgenlab/gonomics/giraf"
+	"github.com/vertgenlab/gonomics/simpleGraph"
+	"io"
+	"log"
+	"strings"
+)
+
+// The BinReader struct wraps the bgzf reader from the biogo repository with a bytes buffer to store encoded giraf records
+type BinReader struct {
+	bg       *bgzf.Reader
+	currData *bytes.Buffer
+}
+
+// NewBinReader creates a new BinReader
+func NewBinReader(file io.Reader) *BinReader {
+	reader, err := bgzf.NewReader(file, 1) //TODO: Play with different levels of concurrency
+	common.ExitIfError(err)
+	return &BinReader{
+		bg:       reader,
+		currData: &bytes.Buffer{},
+	}
+}
+
+// DecompressGiraf will decode a binary giraf file (.giraf.fe) and output a giraf file (.giraf)
+func DecompressGiraf(filename string, graph *simpleGraph.SimpleGraph) {
+	// Initialize infile
+	infile := fileio.EasyOpen(filename)
+	defer infile.Close()
+	reader := NewBinReader(infile.BuffReader)
+	var err error
+
+	// Initialize outfile
+	outfile := fileio.EasyCreate(strings.TrimSuffix(filename, ".fe"))
```

Comment:
What is the significance of .fe for binary? Is this a type of compression, like gz or something else?

Reply:
None whatsoever, this may change later. In general I was thinking it may be best to append a suffix to the filename like gzip instead of sam/bam style. '.fe' was only chosen for grins :)